### ISSUE TITLE
Grad for student forcing in training

### ISSUE THREAD

**Issue by qy-feng** [ruotianluo/ImageCaptioning.pytorch#75]
https://github.com/ruotianluo/ImageCaptioning.pytorch/blob/a1445e48d47d07840f8db4013761e56377ad1b7d/models/ShowTellModel.py#L70

I think what you mean here is disable the back propagation for gradient of it if using student forcing, right?
but the requires_grad of variable `seq` (`labels` in the train.py) is already set to False. 
So if its gradient isn't stopped in the back propagation, will there be a big influence? By the way, I'm using pytorch 0.4.0, can you give me some advice?

**Comment by ruotianluo** [comment]
No worry. I wrote this when pytorch was out for only 1month and I was not familiar with default behavior.
